log/log.go

```diff
@@ -0,0 +1,174 @@
+// Package log Custom logger for labstack/echo that uses sirupsen/logrus.
+package log
+
+import (
+	"io"
+
+	gommonlog "github.com/labstack/gommon/log"
+	"github.com/sirupsen/logrus"
+)
+
+// Logger defines our implementation of Labstack Echo's logger API.
+type Logger struct {
+	Entry *logrus.Entry
+}
+
+// NewLogger creates and returns a new logger.
+func NewLogger(loggerEntry *logrus.Entry) *Logger {
+	return &Logger{Entry: loggerEntry}
+}
+
+// Output returns the output writer associated with the logger.
+func (l *Logger) Output() io.Writer {
+	return l.Entry.Logger.Out
+}
+
+// SetOutput sets the output writer associated with the logger.
+func (l *Logger) SetOutput(writer io.Writer) {
+	l.Entry.Logger.SetOutput(writer)
+}
+
+// Prefix returns the prefix associated with the logger. This is a no-op.
+func (l *Logger) Prefix() string {
+	return ""
+}
+
+// SetPrefix sets the prefix associated with the logger. This is a no-op.
+func (l *Logger) SetPrefix(_ string) {
+}
+
+// Level returns the level associated with the logger. The most restrictive logging level supported by
+// labstack/echo's Logger interface is ERROR. The least restrictive logging level that it supports is
+// DEBUG.
+func (l *Logger) Level() gommonlog.Lvl {
+	if l.Entry.Level == logrus.PanicLevel {
+		return gommonlog.ERROR
+	} else if l.Entry.Logger.Level == logrus.FatalLevel {
+		return gommonlog.ERROR
+	} else if l.Entry.Logger.Level == logrus.ErrorLevel {
+		return gommonlog.ERROR
+	} else if l.Entry.Logger.Level == logrus.WarnLevel {
+		return gommonlog.WARN
+	} else if l.Entry.Logger.Level == logrus.InfoLevel {
+		return gommonlog.INFO
+	} else {
+		return gommonlog.DEBUG
+	}
+}
+
+// SetLevel sets the level associated with the logger. The most restrictive logging level supported by
+// labstack/echo's Logger interface is ERROR. The least restrictive logging level that it supports is
+// DEBUG.
+func (l *Logger) SetLevel(lvl gommonlog.Lvl) {
+	if lvl == gommonlog.ERROR {
+		l.Entry.Logger.SetLevel(logrus.ErrorLevel)
+	} else if lvl == gommonlog.WARN {
+		l.Entry.Logger.SetLevel(logrus.WarnLevel)
+	} else if lvl == gommonlog.INFO {
+		l.Entry.Logger.SetLevel(logrus.InfoLevel)
+	} else {
+		l.Entry.Logger.SetLevel(logrus.DebugLevel)
+	}
+}
+
+// SetHeader adds header fields to log messages. This is a no-op.
+func (l *Logger) SetHeader(_ string) {
+}
```

Comment:
If I'm understanding what this is supposed to do, it might be possible to do something like replace the logrus.Entry (or a second internal one, maybe, to track state if needed?) with a new `logrus.WithFields(...)` when this function is called.

Reply:
The example from the test function appears to interpolate values from the logger definition into the log message:

``` go
func TestFormat(t *testing.T) {
	l := New("test")
	l.SetHeader("${level} | ${prefix}")
	b := new(bytes.Buffer)
	l.SetOutput(b)
	l.Info("info")
	assert.Equal(t, "INFO | test info\n", b.String())
}
```

I hadn't thought about using `WithFields` to implement this, but it seems like it could be done. variables that we're using are different from what `gommon/log` uses, though, so I think I might leave it unimplemented until we have a specific use for it.